triage/2024-11-19.md

```diff
@@ -0,0 +1,268 @@
+
+| (instructions:u)                   | mean  | range          | count |
+|:----------------------------------:|:-----:|:--------------:|:-----:|
+| Regressions ❌ <br /> (primary)    | 0.7%  | [0.5%, 0.9%]   | 2     |
+| Regressions ❌ <br /> (secondary)  | -     | -              | 0     |
+| Improvements ✅ <br /> (primary)   | -0.7% | [-2.0%, -0.2%] | 6     |
+| Improvements ✅ <br /> (secondary) | -     | -              | 0     |
+| All ❌✅ (primary)                 | -0.3% | [-2.0%, 0.9%]  | 8     |
+
+* primary regressions are to image (0.92%) and webrender (0.50%) in opt-full scenarios.
+* it seems like this might be just noise from changes to inlining decisions, since it seems like both those cases are undoing artificial improvements introduced by PR #132870 (noted above)
+* marking as triaged
+
+rustc_metadata: Preprocess search paths for better performance [#132910](https://github.com/rust-lang/rust/pull/132910) [(Comparison Link)](https://perf.rust-lang.org/compare.html?start=f00f68245ebe85960000adaab18c1fe8da104cfe&end=76fd47124b8f8037b6187169b2cdf39139466952&stat=instructions:u)
+
+| (instructions:u)                   | mean  | range          | count |
+|:----------------------------------:|:-----:|:--------------:|:-----:|
+| Regressions ❌ <br /> (primary)    | 0.0%  | [0.0%, 0.0%]   | 1     |
+| Regressions ❌ <br /> (secondary)  | 2.2%  | [0.1%, 4.4%]   | 2     |
+| Improvements ✅ <br /> (primary)   | -     | -              | 0     |
+| Improvements ✅ <br /> (secondary) | -0.2% | [-0.2%, -0.2%] | 2     |
+| All ❌✅ (primary)                 | 0.0%  | [0.0%, 0.0%]   | 1     |
+
+* already marked by saethlin as triaged: "The reported regressions are extremely marginal, and the discussion above: [#132910 (comment)](https://github.com/rust-lang/rust/pull/132910#issuecomment-2473240290) contains instructions for how to demonstrate a quite significant benefit of this change."
+
+Querify MonoItem collection [#132566](https://github.com/rust-lang/rust/pull/132566) [(Comparison Link)](https://perf.rust-lang.org/compare.html?start=5afd5ad29c014de69bea61d028a1ce832ed75a75&end=ee4a56e353dc3ddfcb12df5fe2dc1329a315c2f5&stat=instructions:u)
+
+| (instructions:u)                   | mean  | range           | count |
+|:----------------------------------:|:-----:|:---------------:|:-----:|
+| Regressions ❌ <br /> (primary)    | 1.7%  | [0.3%, 6.6%]    | 10    |
+| Regressions ❌ <br /> (secondary)  | 1.7%  | [0.3%, 3.7%]    | 23    |
+| Improvements ✅ <br /> (primary)   | -6.3% | [-25.1%, -0.1%] | 73    |
+| Improvements ✅ <br /> (secondary) | -9.9% | [-67.2%, -0.3%] | 22    |
+| All ❌✅ (primary)                 | -5.4% | [-25.1%, 6.6%]  | 83    |
+
+* massive improvements; many incremental benchmarks are improved by 5% to 20% (and even 25% in the case of ripgrep)
```

Comment:
this is great no doubt, also remember that changes to incremental data loading/hashing like this are over-represented in rustc-perf results because we force incremental hash verification all the time compared to rustc's 1/32 ratio (for stabilization purposes).